lib/streamlit/elements/deck_gl_json_chart.py

```diff
@@ -33,15 +54,98 @@
 }
 
 
+class LayerSelectionState(TypedDict, total=False):
+    """
+    The schema for the PyDeck Layer Selection State
+
+    Attributes
+    ----------
+    indices : dict[str, list[int]]
+        Dictionary where keys are the layer id and values are lists of indices
+        of selected objects.
+    objects : dict[str, list[dict[str, Any]]]
+        Dictionary where keys are the layer id and values are lists of metadata
+        objects for the selected items.
+    """
+
+    indices: dict[str, list[int]]
+    objects: dict[str, list[dict[str, Any]]]
+
+
+class PydeckState(TypedDict, total=False):
+    """
+    The schema for the PyDeck State
+
+    Attributes
+    ----------
+    selection : LayerSelectionState
+        The selection state of the PyDeck layers.
+    """
+
+    selection: LayerSelectionState
+
+
+@dataclass
+class PydeckSelectionSerde:
+    """PydeckSelectionSerde is used to serialize and deserialize the Pydeck selection state."""
+
+    def deserialize(self, ui_value: str | None, widget_id: str = "") -> PydeckState:
+        empty_selection_state: PydeckState = {
+            "selection": {
+                "indices": {},
+                "objects": {},
+            }
+        }
+
+        selection_state = (
+            empty_selection_state
+            if ui_value is None
+            else cast(PydeckState, AttributeDictionary(json.loads(ui_value)))
+        )
+
+        return selection_state
```

Comment:
I think we need to wrap this in a `AttributeDictionary`. Otherwise if its returning empty state dict, it will just be a normal dict:

```
return cast(PlotlyState, AttributeDictionary(selection_state))
```

Reply:
It is already wrapped in an `AttributeDictionary` in the conditional above, does that look correct to you?